### PR TITLE
Restoring the Test Runtime libraries from a single project.json

### LINKED ISSUE #5174: Change tests to use one common project.json
The CoreCLR tests have many local project.json file. To run the tests, we go over all project.json files and copy the dependencies into a single directory. If two project.json files specify different versions of packages, the last one copied wins that may not be the right one. I have run into many problems with mismatched versions while working on https://github.com/dotnet/coreclr/pull/5154

We should switch to just use one common project.json file instead.

### PR DESCRIPTION
**Pull request by ramarag**
This will ensure the expected version to be present at the runtime

@wtgodbe  @rahku @gkhanna79 @jkotas  PTAL

Fixes #5174

### CODE REVIEW

File: tests/src/runtime/project.json [hunk 1]
```diff
@@ -4,41 +4,72 @@
...
     "test_runtime": {
       "target": "project",
       "exclude": "compile"
     }
   },
   "frameworks": {
     "netcoreapp1.0": {
-      "imports": "dnxcore50"
+      "imports": [
+        "netcoreapp1.0",
```
Comment (jasonwilliams200OK) [inline tests/src/runtime/project.json]: This self-referencing `netcoreapp1.0` TFM is extra, and can safely be removed from the imports list.

**Comment by rahku** [comment]
can't build use the same project.json as runtime

**Comment by rahku** [comment]
@dsgouda this will impact your helix perf targets

**Comment by ramarag** [comment]
Yes we can use the same for the build, i am in the process of testing it out and the change is pretty big [ many files are going away]. Will create a separate PR for it

**Comment by dsgouda** [comment]
We don't have perf test runs working on CoreCLR as of now. Please go ahead and make the necessary changes. We plan  to fix the perf test runs once the functional test runs via helix are enabled.

File: tests/publishdependency.targets [hunk 2]
```diff
@@ -9,23 +9,13 @@
   </ItemGroup>
 
   <ItemGroup>
-    <AllProjectJsonLockFiles Include="$(SourceDir)**\*\project.lock.json"/>
+    <ProjectLockJsonFiles Include="$(SourceDir)\runtime\project.lock.json"/>
```
Comment (gkhanna79) [inline tests/publishdependency.targets]: To confirm, this is being used to restore dependency when creating the overlay, correct?

Comment (ramarag) [inline tests/publishdependency.targets]: yes

**Comment by ramarag** [comment]
cc @dagood  if you see runtime  failure modify the  tests/src/runtime/project.json accordingly

**Comment by dagood** [comment]
Awesome, thanks! Glad to see this change--that test asset resolve step has always been a huge pain to debug.

**Pull request merged**